### PR TITLE
[IMP] util.adapt_domains: handle dotted expressions in `old` argument

### PR DESCRIPTION
**Pull request by KangOl**

### CODE REVIEW

**Comment by KangOl** [comment]
upgradeci retry with always only base hr project

File: src/util/domains.py [hunk 1]
```diff
@@ -196,11 +196,15 @@ def _adapt_one_domain(cr, target_model, old, new, model, domain, adapter=None, f
...
 
+    dot_old = old.split(".")
+    dot_new = new.split(".")
+
     def clean_path(left):
         path = left.split(".")
-        for idx in range(1, len(path) + 1):
-            if path[-idx] == old and _valid_path_to(cr, path[:-idx], model, target_model):
-                path[-idx] = new
+        for idx in range(len(path) - 1, -1, -1):
```
Comment (aj-fuentes) [inline src/util/domains.py:204]:
```suggestion
        for idx in range(len(path) - len(dot_old), -1, -1):
```

**Review by aj-fuentes — COMMENTED** [review]
I'm waiting the whole batch from upgradeci for unforessen issues, but it LGTM already

**Comment by KangOl** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**